tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp

```diff
@@ -1318,6 +1318,45 @@ static std::string GetNavigationId(clang::FunctionDecl const* func)
   return navigationId;
 }
 
+static std::string GetNavigationId(
+    clang::FriendDecl const* friendDecl,
+    std::string const& parentNodeNavigationId)
+{
+  std::string navigationId = parentNodeNavigationId + "_friend_";
```

Comment:
Can the parentNodeNavigationId be ""? Can we assert these premises.

Reply:
It cannot be empty, I'll add an error message if it is.